client/app/pages/data-sources/DataSourcesList.jsx

```diff
@@ -0,0 +1,142 @@
+      loading: false,
+    }, () => { // all resources are loaded in state
+      if ($route.current.locals.isNewDataSourcePage) {
+        this.showCreateSourceDialog();
+      }
+    }));
+  }
+
+  createDataSource = (selectedType, values) => {
+    const target = { options: {}, type: selectedType };
+    helper.updateTargetWithValues(target, values);
+
+    return DataSource.save(target).$promise.then(() => {
+      this.setState({ loading: true });
+      DataSource.query(dataSources => this.setState({ dataSources, loading: false }));
+    }).catch((error) => {
+      if (!(error instanceof Error)) {
+        error = new Error(get(error, 'data.message', 'Failed saving.'));
+      }
+      return Promise.reject(error);
+    });
+  };
+
+  showCreateSourceDialog = () => {
+    CreateSourceDialog.showModal({
+      types: this.state.dataSourceTypes,
+      sourceType: 'Data Source',
+      imageFolder: IMG_ROOT,
+      helpTriggerPrefix: 'DS_',
+      onCreate: this.createDataSource,
+    }).result.then((success) => {
+      if (success) {
+        this.setState({ loading: true });
+        DataSource.query(dataSources => this.setState({ dataSources }));
+      }
+    }).finally(() => {
+      if ($route.current.locals.isNewDataSourcePage) {
+        navigateTo('data_sources');
+      }
+    });
+  };
+
+  renderDataSources() {
+    const { dataSources } = this.state;
+    const items = dataSources.map(dataSource => ({
+      title: dataSource.name,
+      imgSrc: `${IMG_ROOT}/${dataSource.type}.png`,
+      href: `data_sources/${dataSource.id}`,
+    }));
+
+    return isEmpty(dataSources) ? (
+      <div className="text-center">
+        There are no data sources yet.
+        <div className="m-t-5">
+          <a className="clickable" onClick={this.showCreateSourceDialog}>Click here</a> to add one.
+        </div>
+      </div>
+    ) : (<CardsList items={items} />);
+  }
+
+  render() {
+    const newDataSourceProps = {
+      type: 'primary',
+      onClick: policy.isCreateDataSourceEnabled() ? this.showCreateSourceDialog : null,
+      disabled: !policy.isCreateDataSourceEnabled(),
+    };
+
+    return (
+      <div>
+        <div className="m-b-15">
+          <Button {...newDataSourceProps}>
+            <i className="fa fa-plus m-r-5" />
+            New Data Source
+          </Button>
+        </div>
+        {this.state.loading ? <LoadingState className="" /> : this.renderDataSources()}
+      </div>
+    );
+  }
+}
+
+export default function init(ngModule) {
+  settingsMenu.add({
+    permission: 'admin',
+    title: 'Data Sources',
+    path: 'data_sources',
+    order: 1,
+  });
+
+  ngModule.component('pageDataSourcesList', react2angular(DataSourcesList));
+
+  return routesToAngularRoutes([
+    {
+      path: '/data_sources',
+      title: 'Data Sources',
+      key: 'data_sources',
+    },
+    {
+      path: '/data_sources/new',
+      title: 'Data Sources',
+      key: 'data_sources',
+      isNewDataSourcePage: true,
```

Comment:
```suggestion
      isNewDataSourcePage: policy.isCreateDataSourceEnabled(),
```

Reply:
I was actually trying to avoid `/new` being in the url without the dialog open. Perhaps not open the modal and redirect to `/data_sources` in case the user can't create data sources?

Reply:
Go for it